components/frontend/src/utils.test.js

```diff
@@ -259,3 +259,61 @@ it("returns false when the user prefers light mode", () => {
     matchMediaMatches = false
     expect(userPrefersDarkMode(null)).toBe(false)
 })
+
+it("returns the metric response overrun when there are no measurements", () => {
+    expect(getMetricResponseOverrun("uuid", {}, {}, [])).toStrictEqual({"overruns": [], "totalOverrun": 0})
+})
+
+it("returns the metric response overrun when there is no overrun", () => {
+    expect(getMetricResponseOverrun("uuid", {}, {}, [{metric_uuid: "uuid", start: "2000-01-01", end: "2000-01-04"}])).toStrictEqual(
+        {"overruns": [], "totalOverrun": 0}
+    )
+})
+
+it("returns the metric response overrun when there is one long measurement", () => {
+    expect(getMetricResponseOverrun("uuid", {}, {}, [{metric_uuid: "uuid", start: "2000-01-01", end: "2000-01-31"}])).toStrictEqual(
+        {
+            "overruns": [
+                {
+                    "status": "unknown",
+                    "start": "2000-01-01",
+                    "end": "2000-01-31",
+                    "actual_response_time": 30,
+                    "desired_response_time": 3,
+                    "overrun": 27
```

Comment:
hmm, the unit of overrun is the same as actual and desired times.. and the unit of all three is actually days! a little weird, but I guess adding `_days` to each would be even more messy :-)

Reply:
Yeah, I'm not a fan of Hungarian notation. Maybe we should do more with types in JS, but I haven't looked into that yet.